packages/dx-react-scheduler/src/plugins/view-state.jsx

```diff
@@ -24,48 +24,46 @@ export class ViewState extends React.PureComponent {
           const { onCurrentDateChange } = this.props;
           return onCurrentDateChange;
         },
-        currentView: () => {
-          const { onCurrentViewChange } = this.props;
-          return onCurrentViewChange;
+        currentViewName: () => {
+          const { onCurrentViewNameChange } = this.props;
+          return onCurrentViewNameChange;
         },
       },
     );
 
     this.changeCurrentDate = stateHelper.applyFieldReducer
       .bind(stateHelper, 'currentDate', changeCurrentDate);
-    this.setCurrentView = stateHelper.applyFieldReducer
-      .bind(stateHelper, 'currentView', setCurrentView);
+    this.setCurrentViewName = stateHelper.applyFieldReducer
+      .bind(stateHelper, 'currentViewName', setCurrentViewName);
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
     const {
       currentDate = prevState.currentDate,
-      currentView = prevState.currentView,
+      currentViewName = prevState.currentViewName,
     } = nextProps;
 
     return {
       currentDate,
-      currentView,
+      currentViewName,
     };
   }
 
   render() {
-    const { currentDate, currentView: stateCurrentView } = this.state;
-
-    const currentViewComputed = ({ currentView }) => {
-      if (currentView !== stateCurrentView) {
-        if (!currentView) return stateCurrentView;
-        if (!stateCurrentView) return currentView;
-      } return currentView;
-    };
+    const { currentDate, currentViewName: stateCurrentViewName } = this.state;
+    const currentViewComputed = () => (
+      stateCurrentViewName
+        ? { name: stateCurrentViewName }
+        : undefined
```

Comment:
Why you create an object only with a `name` property here, but in views create the object with `name` and `type`?

Reply:
Here, I don't know anything about a `type`.